Source/CarthageKit/Xcode.swift

```diff
@@ -368,15 +369,13 @@ private func mergeBuildProductsIntoDirectory(
 				}
 
 			let sourceModulesURL = SignalProducer(result: secondProductSettings.relativeModulesPath.fanout(secondProductSettings.builtProductsDirectoryURL))
-				.filter { $0.0 != nil }
-				.map { (modulesPath, productsURL) -> URL in
-					return productsURL.appendingPathComponent(modulesPath!)
+				.filterMap { modulesPath, productsURL in
+					return modulesPath.map { productsURL.appendingPathComponent($0) }
 				}
 
 			let destinationModulesURL = SignalProducer(result: firstProductSettings.relativeModulesPath)
-				.filter { $0 != nil }
-				.map { modulesPath -> URL in
-					return destinationFolderURL.appendingPathComponent(modulesPath!)
+				.filterMap { modulesPath in
+					return modulesPath.map { destinationFolderURL.appendingPathComponent($0) }
```

Comment:
These two make sense to me since a separate filter step requires `!`, but I think the other examples are clearer with a separate filter.